LinkDotNet.ValidationExtensions/RangeAttributes/GenericDynamicRangeAttribute.cs

```diff
@@ -0,0 +1,156 @@
+    public DynamicRangeAttribute(T minimum, T maximum)
+        : base()
+    {
+        getMinimum = (ValidationContext validationContext) => minimum;
+        getMaximum = (ValidationContext validationContext) => maximum;
+    }
+
+    /// <inheritdoc />
+    protected override ValidationResult? IsValid(object? value, ValidationContext validationContext)
+    {
+        var operandType = typeof(T);
+        var minimumAsText = getMinimum.Invoke(validationContext).ToString()!;
+        var maximumAsText = getMaximum.Invoke(validationContext).ToString()!;
+
+        // Create RangeAttribute instance for validate value in actual range
+        var rangeAttribute = new RangeAttribute(operandType, minimumAsText, maximumAsText);
+
+        if (ErrorMessage is not null)
+        {
+            rangeAttribute.ErrorMessage = ErrorMessage;
+        }
+
+        if (ErrorMessageResourceName is not null)
+        {
+            rangeAttribute.ErrorMessageResourceName = ErrorMessageResourceName;
+        }
+
+        if (ErrorMessageResourceType is not null)
+        {
+            rangeAttribute.ErrorMessageResourceType = ErrorMessageResourceType;
+        }
+
+        return rangeAttribute.GetValidationResult(value, validationContext);
+    }
+
+    /// <summary>
+    /// Get actual value of subject; subject is 'Minimum' or 'Maximum'.
+    /// </summary>
+    /// <param name="validationContext">
+    /// A <see cref="ValidationContext" /> instance that provides
+    /// context about the validation operation, such as the object and member being validated.
+    /// </param>
+    /// <param name="subjectName"> The name of subject. </param>
+    /// <param name="propertyName"> The propertyName or value of subject. </param>
+    /// <returns> The actual value of subject. </returns>
+    /// <exception cref="ArgumentNullException"> is thrown when propertyNameOrValue is null or nothing. </exception>
+    /// <exception cref="InvalidOperationException"> is thrown when PropertyType of propertyName has not been the same as OperandType.</exception>
+    private static T GetActualValue(ValidationContext validationContext, string subjectName, string propertyName)
+    {
+        if (string.IsNullOrWhiteSpace(propertyName))
+        {
+            throw new InvalidOperationException($"The '{subjectName} PropertyName' cannot be null or empty.");
+        }
+
+        var propertyInfo = validationContext.ObjectType.GetProperty(propertyName);
+
+        if (propertyInfo is null)
+        {
+            throw new InvalidOperationException($"The '{propertyName}' property not found (introduced for '{subjectName}' in range).");
+        }
+
+        var subjectType = typeof(T);
+        var propertyType = propertyInfo.PropertyType;
+        if (propertyType != subjectType)
+        {
+            if (propertyType.IsGenericType && propertyType.GetGenericTypeDefinition() == typeof(Nullable<>))
+            {
+                // it is nullable so we extrcat underlying type
```

Comment:
Small typo - but anyway I'd remove the comment and refactor the `if`:
```csharp
var isNullableType = propertyType.IsGenericType && propertyType.GetGenericTypeDefinition() == typeof(Nullable<>);

if (isNullableType) { ... }
```

Reply:
It was already left from last PR :wink:
It's done [a6b4609](https://github.com/linkdotnet/ValidationExtensions/pull/7/commits/a6b460999f92bfad2c4e4064b6180acc18f3df07)